### ISSUE TITLE
Additional Early Stopping Scheme

### ISSUE THREAD

**Issue by terrytangyuan** [tensorflow/skflow#102]
Currently skflow only supports early stopping scheme that's used mostly frequently by Kagglers, specific code can be found here: https://github.com/tensorflow/skflow/blob/master/skflow/monitors.py#L92

It would also be good allow other schemes such as watching changes in moving averages, etc, since sometimes the current scheme may not be very effective. @ilblackdragon may have more thoughts on this. 

PRs are welcomed.

**Comment by mheilman** [comment]
It looks like the validation monitor checks performance on the validation data after every minibatch currently.  This seems very inefficient if one's validation set is much larger than a minibatch.  Am I missing something?  If not, it'd be nice to have a keyword argument for how frequently to check validation set performance (e.g., so that it could be checked every 100 minibatches or so).

(This seemed the most relevant place to mention this, though it's not exactly the same issue.)

**Comment by ilblackdragon** [comment]
@terrytangyuan Move this to tensorflow/tensorflow?

**Comment by terrytangyuan** [comment]
This will probably be included in new monitor interface. Closing now.
